src/masternodeman.cpp

```diff
@@ -1212,7 +1214,9 @@ void CMasternodeMan::UpdateMasternodeList(CMasternodeBroadcast mnb)
 
 bool CMasternodeMan::CheckMnbAndUpdateMasternodeList(CMasternodeBroadcast mnb, int& nDos)
 {
-    LOCK(cs);
+    // lock cs_vNodes (will be required in mnb.Releay()) early to ensure consistent lock order
+    // during masternode list sync process
+    LOCK2(cs_vNodes, cs);
```

Comment:
> I think it would make more sense and be easier to manage if we lock higher-level mutexes, before lower-level ones like cs_vNodes.

Hmm.. good point 👍